src/lib/operators/multi_predicate_join.hpp

```diff
@@ -0,0 +1,46 @@
+#pragma once
+
+#include <memory>
+#include <vector>
+
+#include "all_type_variant.hpp"
+#include "expression/binary_predicate_expression.hpp"
+#include "expression/evaluation/expression_evaluator.hpp"
+#include "expression/evaluation/expression_result.hpp"
+#include "expression/value_expression.hpp"
+#include "storage/table.hpp"
+#include "types.hpp"
+
+namespace opossum {
+
+inline AllTypeVariant _get_value(const Table& table, const RowID row_id, const ColumnID& column_id) {
+  const auto& segment = *table.get_chunk(row_id.chunk_id)->segments()[column_id];
+  return segment[row_id.chunk_offset];
+}
+
+inline bool _fulfills_join_predicates(const Table& left_table, const Table& right_table, const RowID left_row_id,
+                                      const RowID right_row_id, const std::vector<JoinPredicate>& join_predicates) {
```

Comment:
This method is only used in the `JoinSortMerge` operator and doesn't have to be in the `opossum` namespace. Also, how does this compare to the MultiPredicateJoinEvaluator?

Reply:
This way of predicate evaluation was a solution before we got the concept of the accessors. Since the performance of that old solution is worse than the current way using the MultPredJoinEval, we removed it completely.